AppController/djinn.rb

```diff
@@ -2644,7 +2594,7 @@ def start_new_roles_on_nodes(nodes_needed, instance_type, secret)
     return BAD_SECRET_MSG unless valid_secret?(secret)
 
     # TODO: we should validate the roles type, to ensure they are valid.
-    if nodes_needed.class != Array || instance_type.class != String
+    if nodes_needed.class != Array || instance_type.class != String && is_cloud
```

Comment:
I would recommend adding parentheses for clarity. Though there is no ambiguity for the interpreter, a reader without a firm grasp of operator precedence in Ruby might wonder if the above represents
`if (nodes_needed.class != Array || instance_type.class != String) && is_cloud`
or
`if nodes_needed.class != Array || (instance_type.class != String && is_cloud)`